execute/executor.go

```diff
@@ -297,7 +297,9 @@ func (es *executionState) do(ctx context.Context) {
 			select {
 			case <-t.Finished():
 			case <-ctx.Done():
-				es.abort(errors.New("context done"))
+				if err := ctx.Err(); err != nil {
```

Comment:
Is an error guaranteed not to be nil here?

Reply:
The documentation says that if done is closed then this will always return non-nil so I'll remove the check.